kloppy/infra/serializers/tracking/statsperform.py

```diff
@@ -0,0 +1,271 @@
+        if "xml_metadata" not in inputs:
+            raise ValueError("Please specify a value for 'xml_metadata'")
+        if "raw_data" not in inputs:
+            raise ValueError("Please specify a value for 'raw_data'")
+
+    def deserialize(self, inputs: StatsperformInputs) -> TrackingDataset:
+        raw_data = inputs.raw_data.read()
+        metadata = inputs.meta_data.read()
+
+        with performance_logging("Loading XML metadata", logger=logger):
+            match = objectify.fromstring(metadata).matchInfo
+            teams = []
+            for team_info in match.contestants.iterchildren(tag="contestant"):
+                if team_info.attrib["position"] == "home":
+                    ground = Ground.HOME
+                else:
+                    ground = Ground.AWAY
+                team = Team(
+                    team_id=team_info.attrib["id"],
+                    name=team_info.attrib["name"],
+                    ground=ground,
+                )
+                teams.append(team)
+
+            # Get Frame Rate - Either 10FPS or 25FPS
+            frame_rate = self.__get_frame_rate(raw_data)
+            pitch_size_length = 100
+            pitch_size_width = 100
+
+            periods = self.__get_periods(raw_data)
+            # Get Player Info:
+            for line_up in objectify.fromstring(
+                metadata
+            ).liveData.iterchildren(tag="lineUp"):
+                count = 0
+                for team in teams:
+                    if team.team_id == line_up.get("contestantId"):
+                        team_index = count
+                    count += 1
+
+                for player in line_up.iterchildren(tag="player"):
+                    player = player.attrib
+                    player = Player(
+                        player_id=player["playerId"],
+                        name=player["matchName"],
+                        starting=player["position"] == "substitute",
+                        position=player["position"],
+                        team=team,
+                        jersey_no=player["shirtNumber"],
+                        attributes={},
+                    )
+                    teams[team_index].players.append(player)
+
+        # Handles the tracking frame data
+        with performance_logging("Loading raw data", logger=logger):
+            transformer = self.get_transformer(
+                length=pitch_size_length, width=pitch_size_width
+            )
+
+            def _iter():
+                n = 0
+                sample = 1.0 / self.sample_rate
+
+                for line_ in raw_data.decode("ascii").splitlines():
+                    frame_id = int(line_.split(";")[1].split(",")[0])
+                    if self.only_alive and not line_.endswith("Alive;:"):
+                        continue
+                    period_id = int(line_.split(";")[1].split(",")[1])
+                    period_ = periods[period_id - 1]
```

Comment:
How about letting the `__get_periods` class method return a `dict` instead of a `list` such that the `Period` object could be retrieved by its period ID? I would avoid relying on list indexes whenever possible.

Reply:
Currently, all deserializers are using a `list` for the periods variable. Making the logic to retrieve the first half period as follows: `periods[0]`.

If we would want to change the logic to allow for using a `dict`, I would propose to create a new `issue` to change this for all deserializers.